### PR TITLE
feat: implement byte (de)serialization

### PR DESCRIPTION
**Pull request by remnestal**
Functions for transforming a set into bytes and transforming a byte array into a set.

### CODE REVIEW

File: set_test.go [hunk 1]
```diff
@@ -630,3 +632,96 @@ func TestNextPow2(t *testing.T) {
...
 	}
 }
+
+func TestBytes(t *testing.T) {
+	t.Run("no data", func(t *testing.T) {
+		if b := new(Set).Bytes(); b != nil {
+			t.Errorf("new(Set).Bytes() = %v; want nil", b)
+		}
+	})
+	t.Run("one underkying uint64, where set length is a multiple of 8", func(t *testing.T) {
```
Comment (patstrom) [inline set_test.go]: > underkying

critical typo

File: set_test.go [hunk 2]
```diff
@@ -630,3 +632,96 @@ func TestNextPow2(t *testing.T) {
...
+		s := new(Set).AddRange(0, 64)
+		b := s.Bytes()
+		if !bytes.Equal(b, []byte{255, 255, 255, 255, 255, 255, 255, 255}) {
+			t.Errorf("new(Set).AddRange(0, 64).Bytes() = %v; want [255, ..., 255]", b)
+		}
+	})
+	t.Run("more than one underkying uint64, where set length is a multiple of 8", func(t *testing.T) {
```
Comment (patstrom) [inline set_test.go]: > underkying

critical typo

**Review by patstrom — CHANGES_REQUESTED** [review]

File: set_test.go (hunk 2 above)
Comment (remnestal) [reply set_test.go]: Fixed in 23918b5

File: set_test.go (hunk 1 above)
Comment (remnestal) [reply set_test.go]: Fixed in 23918b5

**Review by patstrom — APPROVED** [review]

**Pull request merged**